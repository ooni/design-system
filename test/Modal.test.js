/* global describe, test, expect, jest */
import React from 'react'
import { fireEvent } from '@testing-library/react'

import { renderWithTheme } from './index'
import { Modal } from '../'

describe('Modal', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <Modal show={false}>Some Content</Modal>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders with close button', () => {
    const { container } = renderWithTheme(
      <Modal closeButton='right' />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('close button click handler is called', () => {
    const onCloseClick = jest.fn()
    const { getByText, getByRole, debug } = renderWithTheme(
      <Modal
        show={true}
        closeButton='right'
        onHideClick={onCloseClick}
      >
        <div> Test </div>
      </Modal>
    )
    expect(getByText('Test')).toBeTruthy()
    const closeButton = getByRole('button')
    fireEvent.click(closeButton)
    expect(onCloseClick).toHaveBeenCalledTimes(1)
  })

})
