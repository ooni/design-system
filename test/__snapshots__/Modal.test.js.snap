// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  display: none;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  max-width: 100vw;
  max-height: 100vh;
  overflow: auto;
  z-index: 1050;
  background-color: #f8f9fa;
  border-radius: radius;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Some Content
  </div>
</div>
`;

exports[`Modal renders with close button 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  display: none;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  max-width: 100vw;
  max-height: 100vh;
  overflow: auto;
  z-index: 1050;
  background-color: #f8f9fa;
  border-radius: radius;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: #f8f9fa;
  background-color: #0588cb;
  border: 0;
  border-radius: 4px;
  display: inline-block;
  padding-left: 16px;
  padding-right: 16px;
  text-transform: capitalize;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  outline: 0;
  color: #adb5bd;
  background-color: transparent;
  border-radius: none;
  border-width: 2px;
  border-style: solid;
  border-color: #0588cb;
  border: none;
  border-width: 2px;
  font-size: 16px;
}

.c2:hover:enabled {
  border-color: #37a6ed;
}

.c2:active:enabled {
  border-color: #0f77b8;
}

.c2:enabled {
  cursor: pointer;
}

.c2:hover {
  color: #ced4da;
}

.c2:active {
  color: #868e96;
}

.c3 {
  position: absolute;
  top: 0;
  height: 28px;
  padding: 0;
  margin: 0;
  left: initial;
  right: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <button
      class="c2 c3"
      font-size="1"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </div>
</div>
`;
